src/logo/logo.re

```diff
@@ -2,98 +2,94 @@ external requestAnimationFrame : (unit => unit) => unit = "" [@@bs.val];
 
 open Constants;
 
-module Logo = {
+type state = {
+  degrees: float,
+  velocity: float,
+  drag: float,
+  lastMs: float
+};
+
+let renderGraphic rotationStyle =>
+  <g fill="none" stroke="none">
+    <g transform="scale(1.5, 1.5) translate(100.000000, 105.000000)">
+      <path fill="rgba(0,0,0,0.1)" d=border_path />
+      <path fill="#dd4b39" d=bg_path />
+      <g transform="translate(55, 29)">
+        <g style=rotationStyle>
+          <path fill="#FFFFFF" d=center_dot_path />
+          <g stroke="#FFFFFF" strokeWidth="8">
+            <path d=ring_one_path />
+            <path d=ring_two_path transform=ring_two_rotate />
+            <path d=ring_three_path transform=ring_three_rotate />
+          </g>
+        </g>
+      </g>
+    </g>
+  </g>;
+
+
+/**
+ * On Mouse Up.
+ */
+let handleMouseUp _event state _self => {
+  let withAccel = state.velocity +. clickAccel;
+  let nextVelocity = withAccel < maxVel ? withAccel : maxVel;
+  ReasonReact.Update {...state, velocity: nextVelocity, drag: mouseUpDrag}
+};
+
 
-  /**
-   * Include Stateful Component Base!
-   */
-  include ReactRe.Component.Stateful;
-  type state = {degrees: float, velocity: float, drag: float, lastMs: float};
-  type props = {message: string};
-  let getInitialState _ /* props */ => {
-    drag: mouseUpDrag,
-    degrees: 0.0,
-    velocity: 0.1,
-    lastMs: Js.Date.now ()
-  };
-  let componentDidMount {setState} => {
+/**
+ * On Mouse Down.
+ */
+let handleMouseDown _event state _self => ReasonReact.Update {...state, drag: mouseDownDrag};
```

Comment:
We've considered having a record argument to things like `render`.

If we did use a record, instead of wrapping self and state in yet another record, perhaps we would make `self` include the `state`.

```
render: fun {state, update} => {

}
```

The only downside here is that `initialState` can never take `self` argument, since state is not yet defined.

Thanks for the feedback/questions.